### ISSUE TITLE
Invade button not displayed for bombarded planets with native species

### ISSUE THREAD

**Issue by dbenage-cx** [freeorion/freeorion#1993]
### Environment

* **FreeOrion Version**: 0.4.7+ 4bcf72b
* **Operating System**: Fedora 26
* **Fetched as**  Compiled from source

### Description
The sidepanel invade button does not display on a bombarded planet containing unowned native, with troop ships in system.

### Expected Result
Invade button is displayed and functional.

### Steps to reproduce
With a target planet containing a native species (and possibly Unowned):
* Attack the planet with armed ships
* When shields of target planet are at 0.0, have ships with troops in system
* Invade button does not display
[Example save](https://github.com/freeorion/freeorion/files/1722096/no_invade.zip) (Albans β at bottom)

See two paths to resolve:
* Ensure `LastTurnBattleHere` / `LastTurnAttackedByShip` checks for current meter gain effects are for previous turn (`... < CurrentTurn - 1` / `Turn low = ... + 1`).
* Change c++ checks in sidepanel.cpp/order.cpp from `CurrentMeterValue` to `InitialMeterValue`

**Comment by geoffthemedio** [comment]
Switched some Current to Initial in master. Do you mean both of the two paths are needed, or both could possibly help?

**Comment by Dilvish-fo** [comment]
> Ensure LastTurnBattleHere / LastTurnAttackedByShip checks for current meter gain effects are for previous turn (... < CurrentTurn - 1 / Turn low = ... + 1)

It looks to me like those two forms are not really equivalent.  ```Turn low = ... + 1``` would be fine, but I can't think of any situation in which we'd want ```... < CurrentTurn - 1```   In fact, that was a bug that we just fixed, dropping the -1 in that situation.  Keep in mind that combat precedes Effect application in turn processing.  ```... <= CurrentTurn - 1``` could be fine, but it is simpler to just have ```... < CurrentTurn```

**Comment by Dilvish-fo** [comment]
> Switched some Current to Initial in master.

@geoffthemedio I've looked over that commit, and the change from current meter to initial meter is puzzling me.  If there is some reason that the CurrentMeter reading would be invalid, please explain, so that I can figure out if it could be affecting the AI the same way you expect it might be affecting the player UI (it seems likely it would). 

But also, regardless, I don't see under what situation those particular changes would impact the presentation of the invade button.

@dbenage-cx you also were thinking that a change from CurrentMeter to InitialMeter (albeit in a different place than Geoff's change) could resolve the problem-- could you please explain your thinking on that, again so I can consider relevance to the AI.

(The AI always uses CurrentMeter checks, never looking at InitialMeter values.)

**Comment by geoffthemedio** [comment]
Should probably have changed all the `CurrentMeterValue` calls here as well: https://github.com/freeorion/freeorion/blob/master/UI/SidePanel.cpp#L1593

`CurrentMeterValue` returns the meter values that are being estimated given that a combat hasn't (yet) occurred on the current turn, so the combat-dependent meters will be expected to increase, thus disallowing the invasion that requires certain meters to be 0. `InitialMeterValue` returnst he meter values as of the start of the current turn, which is before they are being re-estimated.

**Comment by dbenage-cx** [comment]
> Do you mean both of the two paths are needed...?

(~Yes~ Edit: either the scripts or both c+ files need change)
When the issue was posted (I have not checked current master yet), the values for `METER_SHIELD` of Albens β I is initial: 0.0 current: 1.0

`shielded` at [UI/SidePanel.cpp:1596](https://github.com/freeorion/freeorion/blob/master/UI/SidePanel.cpp#L1596) was set true, preventing button from displaying.

Other methods of issuing an invade order would also fail from the check at [util/Order.cpp:628](https://github.com/freeorion/freeorion/blob/master/util/Order.cpp#L628)

Similarly, colonize/bombard and possibly other meter checks would need adjustment if this approach is taken (not verified, UrshMost [noted](http://freeorion.org/forum/viewtopic.php?f=28&t=10878#p91567) ships are repairing during combat)

After verifying master, I will post a diff for the alternative of changing scripts, the resolution/cause to #1986 is not clear to me.

**Comment by dbenage-cx** [comment]
Issue still present with master (bdd8158), alternative fix only for this specific issue:
```diff
diff --git a/default/scripting/species/common/planet_shields.macros b/default/scripting/species/common/planet_shields.macros
index 8bfbd3a1b..41a426737 100644
--- a/default/scripting/species/common/planet_shields.macros
+++ b/default/scripting/species/common/planet_shields.macros
@@ -23,7 +23,7 @@ STANDARD_SHIELD_GROWTH
                 Planet
                 Unowned
                 (Source.LastTurnConquered < CurrentTurn)
-                (Source.System.LastTurnBattleHere < CurrentTurn)
+                (Source.System.LastTurnBattleHere < CurrentTurn - 1)
             ]
             priority = [[AFTER_ALL_TARGET_MAX_METERS_PRIORITY]]
             effects = SetShield value = min(Value(Target.MaxShield), Value + 1)
```

**Comment by geoffthemedio** [comment]
Pushed some additional changes that should help. At least it did for my tests.

**Comment by Dilvish-fo** [comment]
> CurrentMeterValue returns the meter values that are being estimated given that a combat hasn't (yet) occurred on the current turn

@geoffthemedio I take it this is a new thing for CurrentMeter, as part of the shift to meter growth to Effects?  We certainly didn't used to have this issue.   Is it really necessary to have CurrentMeters be making this "estimate"?  That seems both to largely detract from the usefulness of moving meter growth into Effects, and to make the CurrentMeter values basically useless for the UI and for the AI.  At least for the purposes we see here.   I am trying to think through now when we would need to use InitialMeter values and when we'd need to use CurrentMeter values.    Could you do some writeup of that on the forum?

**Comment by geoffthemedio** [comment]
The change has more to do with getting rid of the false "Unknown" meter discrepancies, which happened at the same time as moving meter changes into effects.

`InitialMeterValue` tells you the meter value for this turn, which is what determines what will happen when the turn cycles. `CurrentMeterValue` is an estimate of what it will be next turn, which will change depending on focus changes. This determines how the meter bars update on the planet panel when focus is changed, making it very useful for the UI.

(Internally, `CurrentMeterValue` is also used as the accumulator as effects act on meters and combine to produce the final values.)

Basically, use `InitialMeterValue` for almost everything in the AI, except if you want to get an estimate of a meter on the next turn after the one being played.

**Comment by dbenage-cx** [comment]
@geoffthemedio Thanks for addressing

Fixed by ac0b7f5
